frontend/integration-tests/views/dashboard.view.ts

```diff
@@ -11,3 +11,10 @@ export const activityCard = $('[data-test-id="activity-card"]');
 export const eventsPauseButton = $('[data-test-id="events-pause-button"]');
 export const launcherCard = $('[data-test-id="launcher-card"]');
 export const resourceQuotasCard = $('[data-test-id="resource-quotas-card"]');
+export const insightsButton = element(by.xpath('.//button[normalize-space(.)="Insights"]'));
+export const insightsPopover = element(
+  by.xpath('.//div[@role="dialog" and .//h6[normalize-space(.)="Insights status"]]'),
```

Comment:
Please avoid complicated xpath selectors in tests. It's much better to add a `data-test` attribute to the element and use that.

Reply:
@spadgett there is no way to add such attributes to these elements without patching patternfly-core.

Reply:
> @spadgett there is no way to add this attribute to these elements without patching patternfly-core.

Is that true? `headerContent` on the `Popover` takes a DOM node, so we could pass a `span` with a `data-test` attribute.